### ISSUE TITLE
Failed to start audio capture / Demuxer 'audio': stream explicitly disabled by the device on Android 11 

### ISSUE THREAD

**Issue by PureIncompetence** [Genymobile/scrcpy#4147]
- [x] I have read the [FAQ](https://github.com/Genymobile/scrcpy/blob/master/FAQ.md).
 - [x] I have searched in existing [issues](https://github.com/Genymobile/scrcpy/issues).

 - OS: Ubuntu
 - scrcpy version: 2.1
 - installation method: manual build
 - device model: ASUS
 - Android version: 11 (LOS-based Custom ROM)

Sound does not work by default. As I've read in several other issues regarding sound, you bring up the Shell Heap Dump report activity for a short period of time to make android (11) think that scrcpy - which has the same uid - would be running upfront. That does not work on my device (ofc. with the screen on).
If I, however, run `adb shell "am start com.android.shell/.HeapDumpActivity"` before I run scrcpy, it does work. (The Heap Dump report activity pops up, scrcpy starts, closes it and audio works as expected.) Below I will post the output of what scrcpy logs without the manually started activity:
```
scrcpy 2.1 <https://github.com/Genymobile/scrcpy>
scrcpy-server-v2.1: 1 file pushed. 1.4 MB/s (56955 bytes in 0.040s)
[server] INFO: Device: [asus] asus ASUS_[redacted] (Android 11)
INFO: Renderer: opengl
INFO: OpenGL version: 4.6 (Compatibility Profile) Mesa 23.1.3
INFO: Trilinear filtering enabled
INFO: Texture: 1080x2336
[server] ERROR: Failed to start audio capture
[server] ERROR: On Android 11, audio capture must be started in the foreground, make sure that the device is unlocked when starting scrcpy.
WARN: Demuxer 'audio': stream explicitly disabled by the device
```

When searching for similar Issues, I found #3796 which looked a lot like my issue. I therefore tried to apply the same fix which was successfull there: Setting `SystemClock.sleep()` to a higher value (I tried `500` ms). After that I recompiled the scrcpy-server according to https://github.com/Genymobile/scrcpy/blob/master/doc/build.md#option-1-build-everything-from-sources (which is a real PITA btw, thank you for delivering that one pre-build by default.) and didnt notice any difference. On the other hand I have no experience with java whatsoever so I might have done some mistakes when recompiling.

**Comment by rom1v** [comment]
On `dev` branch, try with:

```diff
diff --git a/server/src/main/java/com/genymobile/scrcpy/AudioCapture.java b/server/src/main/java/com/genymobile/scrcpy/AudioCapture.java
index 7b20cce4f..8b30d6f26 100644
--- a/server/src/main/java/com/genymobile/scrcpy/AudioCapture.java
+++ b/server/src/main/java/com/genymobile/scrcpy/AudioCapture.java
@@ -118,7 +118,7 @@ public final class AudioCapture {
         if (Build.VERSION.SDK_INT == Build.VERSION_CODES.R) {
             startWorkaroundAndroid11();
             try {
-                tryStartRecording(3, 100);
+                tryStartRecording(6, 100);
             } finally {
                 stopWorkaroundAndroid11();
             }
```

and run with `scrcpy -Vdebug`. What is the result?

**Comment by rom1v** [comment]
OK, so I will change from 3 to 5 attempts (so the last one is after ~500ms).

**Comment by rom1v** [comment]
Merged into `dev`: 01d785d9a3407dc0389abee688f1a58a5e4f8923
